src/django/api/processing.py

```diff
@@ -639,13 +639,21 @@ def handle_external_match_process_result(id, result, request, should_create):
     if f_l_item is None:
         # Timeout
         return get_error_match_result(id, result)
-    queryset_f_m = FacilityMatchTemp.objects.filter(
+    if should_create:
+        match_object_type = FacilityMatch
+    else:
+        match_object_type = FacilityMatchTemp
+    queryset_f_m = match_object_type.objects.filter(
         facility_list_item=f_l_item.id)
     if queryset_f_m.count() == 0:
         # No Match and Geocoder Returned No Results
         return get_error_match_result(f_l_item.id, result)
-    if (queryset_f_m.count() == 1
-            and queryset_f_m[0].status == FacilityMatchTemp.AUTOMATIC):
+
+    automatic_matches = [
+        match for match in queryset_f_m
+        if match.status == match_object_type.AUTOMATIC
+    ]
+    if (len(automatic_matches) == 1):
```

Comment:
Question: If there is at least 1 automatic match here, shall we operate with new facility here at all? I mean, why do we need `# New Facility` block here?

Reply:
@VadimKovalenkoSNF Yes, Automatic Match status apply for both situations **NEW_FACILITY** & **MATCHED**.